plugins/disabled-ZeronameLocal/SiteManagerPlugin.py

```diff
@@ -4,23 +4,57 @@
 from Debug import Debug
 from http.client import HTTPSConnection, HTTPConnection, HTTPException
 from base64 import b64encode
+from pathlib import Path
+from .NamecoinUtils import is_namecoin_installed, is_namecoin_running, get_namecoin_conf, check_namecoin_rpc_conf, start_namecoin, kill_namecoin
 
 allow_reload = False # No reload supported
 
 @PluginManager.registerTo("SiteManager")
 class SiteManagerPlugin(object):
+    def __del__(self):
+        if self.namecoin_pid:
+            kill_namecoin(self.namecoin_pid)
+            self.log.debug("Kill namecoin pid : {}".format(self.namecoin_pid))
+            return
+
     def load(self, *args, **kwargs):
         super(SiteManagerPlugin, self).load(*args, **kwargs)
-        self.log = logging.getLogger("ZeronetLocal Plugin")
+        self.log = logging.getLogger("ZeronameLocal Plugin")
         self.error_message = None
-        if not config.namecoin_host or not config.namecoin_rpcport or not config.namecoin_rpcuser or not config.namecoin_rpcpassword:
+        self.namecoin_pid = None
+        self.config = dict({
+            'rpcconnect': config.namecoin_host,
+            'rpcport': config.namecoin_rpcport,
+            'rpcuser': config.namecoin_rpcuser,
+            'rpcpassword': config.namecoin_rpcpassword,
+            'path': config.namecoin_path
+        })
+        self.log.debug("Namecoin config through CLI ? {}".format(self.config))
+        self.log.debug("Namecoin installed ? {}".format(is_namecoin_installed()))
+        if is_namecoin_installed():
```

Comment:
If we are going for now only support linux we can check first that the platform is linux before checking `is_namecoin_installed`.

```python
if sys.platform.startswith('linux') and is_namecoin_installed():
```

Reply:
indeed, it is preferable 5c2911e028665b60a6349115f073a66c9f09e2d9